### PR TITLE
[MF-404] Declare extension slot

### PR DESCRIPTION
**Pull request by samuelmale**
Ticket: https://issues.openmrs.org/browse/MF-404

### CODE REVIEW

**Comment by samuelmale** [comment]
@FlorianRappl how do we resolve this typescript error: https://github.com/openmrs/openmrs-esm-patient-registration/pull/64/checks?check_run_id=1806088957#step:7:11?

File: src/patient-registration/patient-registration.component.tsx [hunk 1]
```diff
@@ -108,6 +109,14 @@ interface AddressValidationSchemaType {
   regexFormat: string;
 }
 
+interface CapturePhotoProps {
+  base64EncodedImage: string;
+  imageFile: File;
+  saveAttachment: Function;
```
Comment (FlorianRappl) [inline src/patient-registration/patient-registration.component.tsx]: Can we be a bit more specific here?

File: src/patient-registration/patient-registration.resource.tsx [hunk 2]
```diff
@@ -132,3 +132,51 @@ export function saveRelationship(abortController: AbortController, relationship:
...
+  fileCaption: string,
+  abortController: AbortController,
+  base64Content: string,
+  url: string,
+  obsDatetime: string,
+  concept: string,
+) {
+  const formData = new FormData();
+  const json = {};
+  json['person'] = patientUuid;
```
Comment (FlorianRappl) [inline src/patient-registration/patient-registration.resource.tsx]: These can be initialized directly, i.e.,

```js
const json = {
  person: patientUuid,
  ...
};
```

if you don't want that type then declare it a `Record<string, any>`, such as `const json: Record<string, any> = ...`.

File: src/patient-registration/patient-registration.resource.tsx [hunk 3]
```diff
@@ -132,3 +132,51 @@ export function saveRelationship(abortController: AbortController, relationship:
...
+
+function dataURItoFile(dataURI) {
+  const byteString = atob(dataURI.split(',')[1]);
+  const mimeString = dataURI
+    .split(',')[0]
+    .split(':')[1]
+    .split(';')[0];
+  // write the bytes of the string to a typed array
+  const buffer = new Uint8Array(byteString.length);
+  for (var i = 0; i < byteString.length; i++) {
```
Comment (FlorianRappl) [inline src/patient-registration/patient-registration.resource.tsx]: let

File: src/patient-registration/patient-registration.resource.tsx [hunk 4]
```diff
@@ -132,3 +132,51 @@ export function saveRelationship(abortController: AbortController, relationship:
...
+  formData.append('json', JSON.stringify(json));
+
+  return openmrsFetch(url, {
+    method: 'POST',
+    signal: abortController.signal,
+    body: formData,
+  });
+}
+
+function dataURItoFile(dataURI) {
```
Comment (FlorianRappl) [inline src/patient-registration/patient-registration.resource.tsx]: Type? `string` I guess?

**Review by FlorianRappl — COMMENTED** [review]
On a good way.

Regarding the TypeScript error: Seems to me you are mixing dependencie.s Please update *all* esm-core related packages and make sure no artifacts remain.

Best way to update:

```js
rm -rf package-lock.json node_modules
npm install
``` 

This recreates the node modules and the package lock. It should come out in a more consistent and up-to-date state.

File: src/patient-registration/section/demographics/demographics-section.component.tsx [hunk 5]
```diff
@@ -24,12 +24,11 @@ export const DemographicsSection: React.FC<DemographicsSectionProps> = ({
   const { t } = useTranslation();
   const [field, meta] = useField('addNameInLocalLanguage');
 
-  const onCapturePhoto = (dataUri, selectedFile, saveImage, obsDate, concept) => {
+  const onCapturePhoto = (dataUri, selectedFile, obsDate, concept) => {
```
Comment (FlorianRappl) [inline src/patient-registration/section/demographics/demographics-section.component.tsx]: Here some typing is missing, or?

**Review by FlorianRappl — COMMENTED** [review]
Great changes - thanks!

**Review by FlorianRappl — APPROVED** [review]
All good - thanks!

**Pull request merged**